src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala

```diff
@@ -0,0 +1,142 @@
+package org.downtowndailybread.bethsaida.request
+
+import java.sql.{Connection, ResultSet}
+import java.time.{OffsetDateTime, ZonedDateTime}
+import java.util.UUID
+
+import org.downtowndailybread.bethsaida.Settings
+import org.downtowndailybread.bethsaida.exception.event.EventNotFoundException
+import org.downtowndailybread.bethsaida.model.{Event, EventAttribute, HoursOfOperation, InternalUser}
+import org.downtowndailybread.bethsaida.request.util.{BaseRequest, DatabaseRequest}
+import org.downtowndailybread.bethsaida.service.UUIDProvider
+
+class EventRequest(conn: Connection, settings: Settings) extends BaseRequest
+  with DatabaseRequest with UUIDProvider {
+
+  def getAllServiceEvents(serviceId: UUID): Seq[Event] = {
+    getAllEventsInternal(Some(serviceId), None)
+  }
+
+  def getEvent(serviceId: UUID, eventId: UUID): Event = {
+    getAllEventsInternal(Some(serviceId), Some(eventId)) match {
+      case e :: Nil => e
+      case _ => throw new EventNotFoundException
+    }
+  }
+
+  def createEvent(serviceId: UUID, event: EventAttribute)(implicit iu: InternalUser): UUID = {
+    val eventId = getUUID()
+    val sql =
+      s"""
+         |insert into event (id, service_id, metadata_id)
+         |values (cast(? as uuid), cast(? as uuid), ?)
+       """.stripMargin
+
+    val ps = conn.prepareStatement(sql)
+    ps.setString(1, eventId)
+    ps.setString(2, serviceId)
+    ps.setInt(3, insertMetadataStatement(conn, true))
+    insertEventAttributeInternal(eventId, event, true)
```

Comment:
Do both of these functions insert into event, or am I going crazy?

Reply:
it's a two part process: the master record which does the relational bit, and then the event_attribute info which has the actual data in it.